custom_components/dahua/light.py

```diff
@@ -202,6 +205,61 @@ async def async_turn_off(self, **kwargs):
         await self._coordinator.async_refresh()
 
 
+class AmcrestFloodLight(DahuaBaseEntity, LightEntity):
+    """
+        Representation of a Amcrest Flood Light (for cameras that have them)
+        Unlike the 'Dahua Illuminator', Amcrest Flood Lights do not play nicely
+        with adjusting the 'White Light' brightness.
+    """
+
+    def __init__(self, coordinator: DahuaDataUpdateCoordinator, entry, name):
+        super().__init__(coordinator, entry)
+        self._name = name
+        self._coordinator = coordinator
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return self._coordinator.get_device_name() + " " + self._name
+
+    @property
+    def unique_id(self):
+        """
+        A unique identifier for this entity. Needs to be unique within a platform (ie light.hue). Should not be configurable by the user or be changeable
+        see https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements
+        """
+        return self._coordinator.get_serial_number() + "_flood_light"
+
+    @property
+    def is_on(self):
+        """Return true if the light is on"""
+        return self._coordinator.is_amcrest_flood_light_on()
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return DAHUA_SUPPORTED_OPTIONS
```

Comment:
Since we aren't using "brightness" you can just pass 0 here to disable that in the UI.